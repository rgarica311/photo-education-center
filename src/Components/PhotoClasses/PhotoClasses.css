.classesContainer {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin-top: 100px;
  padding: 50px;
}
