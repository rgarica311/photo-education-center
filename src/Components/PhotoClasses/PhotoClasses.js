import React, { Component } from 'react'
import ClassCard from '../ClassCard/ClassCard'
import './PhotoClasses.css'

export default class BeginnerClasses extends Component {
  constructor(props){
    super(props)
    this.state = {
      data: props.data,
      titles: props.titles,
      descs: props.descs
    }
  }

  render(){
    console.log('this.state.titles', this.state.titles)
    return(
      <>
      <div className='classesContainer'>
        {
          Object.keys(this.state.titles).map(key =>  (
            <ClassCard title={this.state.titles[key].title}
                       desc={this.state.descs[key].class_desc}
                       path={`/${this.state.titles[key].title.split(' ').join('-')}`}
                       history={this.props.history}
                       key={this.state.titles[key].title}/>
          ))
        }
      </div>
      </>
    )
  }
}
