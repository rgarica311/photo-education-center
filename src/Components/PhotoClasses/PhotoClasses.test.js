import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PhotoClasses from './PhotoClasses';

describe(`PhotoClasses Component`, () => {
  const data = {
    col1: 'test data'
  }

  const titles = {
    0: {
      title: 'test title'
    }
  }

  const descs = {
    0: {
      class_desc: 'test desc'
    }
  }
  it('renders PhotoClasses by default', () => {
    const wrapper = shallow(<PhotoClasses data={data} titles={titles} descs={descs}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
