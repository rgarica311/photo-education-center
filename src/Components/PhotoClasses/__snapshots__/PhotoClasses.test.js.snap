// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoClasses Component renders PhotoClasses by default 1`] = `
<Fragment>
  <div
    className="classesContainer"
  >
    <ClassCard
      desc="test desc"
      key="test title"
      path="/test-title"
      title="test title"
    />
  </div>
</Fragment>
`;
